File: LeetCode/Explore/TopInterview/Amazon/DynamicalProgramming/53_maximum_subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
      curVal, maxSum = nums[0], nums[0]
      
      for i in range(1, len(nums), 1):
        curVal = max(nums[i], curVal + nums[i])
        maxSum = max(maxSum, curVal)
        
      return maxSum

File: LeetCode/Explore/TopInterview/Amazon/DynamicalProgramming/53_maximum_subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
      int curVal = nums[0], maxSum = nums[0];
      for (int i = 1; i < nums.size(); i++) {
        curVal = max(nums[i], curVal + nums[i]);
        maxSum = max(maxSum, curVal);
      }
      return maxSum;
    }
};
